cloudinit/sources/DataSourceAzure.py

```diff
@@ -1945,6 +1948,29 @@ def load_azure_ds_dir(source_dir):
     return (md, ud, cfg, {"ovf-env.xml": contents})
 
 
+def configure_ephemeral_and_hotplug_device(netconfig: dict) -> dict:
+    """
+    Configure the ephemeral and hotplug details.
+    parameters:
+        netconfig: dict, Network config details with other ethernet devices.
+
+    returns the updated network configuration
+    """
+    netconfig["ethernets"]["ephemeral"] = {
+        "dhcp4": True,
+        "dhcp4-overrides": {"use-dns": False},
+        "optional": True,
+        "match": {"driver": "hv_netvsc", "name": "!eth0"},
+    }
+
+    netconfig["ethernets"]["hotpluggedeth0"] = {
```

Comment:
How about naming the ethernets in the configuration `primary` and `secondary` instead of `hotpluggedeth0` and `ephemeral`? I believe that the former better expresses the relationship between the two. That said, I am open to other suggestions.